packages/nx/src/command-line/init/command-object.ts

```diff
@@ -1,90 +1,12 @@
-import { Argv, CommandModule } from 'yargs';
-import { parseCSV } from '../yargs-utils/shared-options';
-import { readNxJson } from '../../config/nx-json';
-
-const useV2 =
-  process.env['NX_ADD_PLUGINS'] !== 'false' &&
-  readNxJson().useInferencePlugins !== false;
+import type { CommandModule } from 'yargs';
 
 export const yargsInitCommand: CommandModule = {
   command: 'init',
   describe:
     'Adds Nx to any type of workspace. It installs nx, creates an nx.json configuration file and optionally sets up remote caching. For more info, check https://nx.dev/recipes/adopting-nx.',
-  builder: (yargs) => withInitOptions(yargs),
+  builder: async (yargs) => (await import('./init')).withInitOptions(yargs),
```

Comment:
This change shouldn't make much functional difference, and I'd prefer to keep builder inside command-object. The builders have to run for every command anyway and the idea of the dynamic require on the handler is to avoid pulling in things required for implementation on every command. Having this here defeats that purpose.

If you wanted to pull builder logic / withInitOptions into a separate file, it'd need to be a different file than the implementation logic, and I don't think its worth it. It's not like the yargs stuff is pulling in more than any other command.